tracing/src/main/java/com/palantir/remoting2/tracing/Tracers.java

```diff
@@ -96,11 +100,14 @@ public static Runnable wrap(Runnable delegate) {
         @Override
         public V call() throws Exception {
             Trace originalTrace = Tracer.copyTrace();
+            String originalMdcTraceIdValue = MDC.get(MDC_KEY);
             Tracer.setTrace(trace);
+            MDC.put(MDC_KEY, trace.getTraceId());
```

Comment:
I'm not sure how logback's Converter class could be used to provide a more structured object, as it only has access to the ILoggingEvent. So we could use it to provide more structured access to the MDC, but I'm not seeing how we can avoid it the MDC altogether.